Source/WebCore/platform/soup/PublicSuffixStoreSoup.cpp

```diff
@@ -62,13 +62,13 @@ String PublicSuffixStore::platformTopPrivatelyControlledDomain(StringView domain
 {
     // This function is expected to work with the format used by cookies, so skip any leading dots.
     unsigned position = 0;
-    while (domain[position] == '.')
+    while (position < domain.length() && domain[position] == '.')
```

Comment:
I like this approach more! It feels more lightweight and remains  very readable :-)